src/core/state/state.py

```diff
@@ -39,8 +39,17 @@ class State:
     z: float = 0.0
 
     force_propulsion_thrusters: float = 0.0
-    fuel_mass: float = 0.0
-    chamber_temp: float = 0.0
+    fuel_mass: float = 0.0 # mass of water
```

Comment:
I don't know if the prop architecture has changed since I graduated, but from what I remember there are three states that we need to keep track of in the sim:
- mass of liquid water in the propellant tank
- mass of gaseous H2/O2 in the propellant tank
- Mass of gaseous H2/O2 in the combustion chamber

I hope there's some better documentation than a review comment by a graduated student, but the flow of propellant through the propulsion systems goes like so:

Inside the propellant tank:
- liquid water gets electrolyzed into gaseous h2/o2. 
- the gaseous h2/o2 bubbles up through the water into an open space above the surface of the water, but still in the propellant tank (called the "ullage")

Between propellant tank and combustion chamber:
- the gas in the propellant tank flows through a one-way check valve into the combustion chamber until the pressure in the propellant tank equals the pressure in the combustion chamber. 

Inside the combustion chamber:
- gas accumulates until we stop electrolyzing
- once we fire the thruster, the mass of gas in the combustion chamber gets expelled into space (i.e. goes to 0 from the spacecraft's perspective). The mass of the liquid and gaseous water in the propellant tank remains unchanged through this event.

Sorry if this is way too ramble-y. Double check with prop team first to make sure that the model you're writing is representative of the prop system.